docs/terminology.md

```diff
@@ -0,0 +1,52 @@
+    - OpenXLA is an open ecosystem of performant, portable, and extensible machine learning (ML) infrastructure
+      components that simplify ML development by defragmenting the tools between frontend frameworks and hardware
+      backends. It includes the XLA compiler, StableHLO, VHLO, [PJRT](https://openxla.org/xla/pjrt/overview) and other
+      components.
+- **XLA**
+    - XLA (Accelerated Linear Algebra) is an open source compiler for machine learning. The XLA compiler takes models
+      from popular frameworks such as PyTorch, TensorFlow, and JAX, and optimizes the models for high-performance
+      execution across different hardware platforms including GPUs, CPUs, and ML accelerators. The XLA compiler outputs
+      some code to LLVM, some to "standard" MLIR, and some to [Triton MLIR](https://triton-lang.org/main/dialects/dialects.html)
+      that is processed by (MLIR-based) OpenAI Triton compiler.
+- **PJRT**
+    - [PJRT](https://github.com/openxla/xla/blob/main/xla/pjrt/c/pjrt_c_api.h) is a uniform Device API that simplifies
+      the growing complexity of ML workload execution across hardware and frameworks. It provides a hardware and
+      framework independent interface for compilers and runtimes.
+- **StableHLO**
+    - StableHLO the public interface to OpenXLA, it is a standardized MLIR dialect that may be used by different
+      frameworks and compilers in the OpenXLA ecosystem. XLA supports StableHLO, and immediately converts it to HLO on
+      the input. There are some [StableHLO to StableHLO](https://openxla.org/stablehlo/generated/stablehlo_passes)
+      passes that use MLIR framework work. It is also possible to convert StableHLO to other compilers' IR without using
+      HLO, for example in cases where an existing IR is more appropriate.
+- **CHLO**
+    - CHLO is a collection of higher level operations which are optionally decomposable to StableHLO.
+- **VHLO**
+    - The [VHLO Dialect](https://openxla.org/stablehlo/vhlo) is a MLIR dialect that is a compatibility layer on top of
+      StableHLO. It provides a snapshot of the StableHLO dialect at a given point in time by versioning individual
+      program elements, and is used for serialization and stability.
+- **MHLO**
+    - MHLO aka [MLIR-HLO](https://github.com/tensorflow/mlir-hlo) is a standalone MLIR-based input of XLA, but is
+      deprecated. Users are encouraged to use StableHLO instead.
+- **HLO**
+    - HLO is an internal graph representation (IR) for the XLA compiler (and also supported input). It is **not** based
+      on MLIR, and has it's own textual syntax and binary (protobuf based) representation.
+- **MLIR**
+    - [MLIR](https://mlir.llvm.org) is intended to be a hybrid IR which can support multiple different requirements in a
+      unified infrastructure. It's accepted by LLVM as an input but it is not a language in itself; it is a framework
+      that allows you to define your own dialect, and then compile it to LLVM code.
```

Comment:
No need to mention LLVM input here, since we just want to give the gist of MLIR as it pertains to StableHLO/XLA, how about the following, then we can delete the Dialects portion as well:

MLIR is a hybrid IR infrastructure that allows users to define "dialects" of operations at varying degrees of abstraction, and gradually lower between these opsets performing transformations at each granularity.

Reply:
Updated, let me know if this is reasonable.